server/protocols/webadmin/webadmin-data/src/main/java/org/apache/james/webadmin/routes/AliasRoutes.java

```diff
@@ -0,0 +1,157 @@
+            .message("An alias source needs to be specified in the path")
+            .haltError();
+    }
+
+    @PUT
+    @Path(ROOT_PATH + "/{" + ALIAS_DESTINATION_ADDRESS + "}/sources/{" + ALIAS_SOURCE_ADDRESS + "}")
+    @ApiOperation(value = "adding a source address into an alias")
+    @ApiImplicitParams({
+        @ApiImplicitParam(required = true, dataType = "string", name = ALIAS_DESTINATION_ADDRESS, paramType = "path",
+            value = "Destination mail address of the alias. Sending a mail to the alias source address will send it to " +
+                "that email address.\n" +
+                MAILADDRESS_ASCII_DISCLAIMER),
+        @ApiImplicitParam(required = true, dataType = "string", name = ALIAS_SOURCE_ADDRESS, paramType = "path",
+            value = "Source mail address of the alias. Sending a mail to that address will send it to " +
+                "the email destination address.\n" +
+                MAILADDRESS_ASCII_DISCLAIMER)
+    })
+    @ApiResponses(value = {
+        @ApiResponse(code = HttpStatus.NO_CONTENT_204, message = "OK"),
+        @ApiResponse(code = HttpStatus.BAD_REQUEST_400, message = ALIAS_DESTINATION_ADDRESS + " or alias structure format is not valid"),
+        @ApiResponse(code = HttpStatus.BAD_REQUEST_400, message = "The alias source exists as an user already"),
+        @ApiResponse(code = HttpStatus.INTERNAL_SERVER_ERROR_500,
+            message = "Internal server error - Something went bad on the server side.")
+    })
+    public HaltException addToAliasSources(Request request, Response response) throws UsersRepositoryException, RecipientRewriteTableException {
+        MailAddress aliasSourceAddress = parseMailAddress(request.params(ALIAS_SOURCE_ADDRESS));
+        ensureUserDoesNotExist(aliasSourceAddress);
+        MailAddress destinationAddress = parseMailAddress(request.params(ALIAS_DESTINATION_ADDRESS));
+        MappingSource source = MappingSource.fromUser(User.fromLocalPartWithDomain(destinationAddress.getLocalPart(), destinationAddress.getDomain()));
+        addAlias(source, aliasSourceAddress);
+        return halt(HttpStatus.NO_CONTENT_204);
+    }
+
+    private void addAlias(MappingSource source, MailAddress aliasSourceAddress) throws RecipientRewriteTableException {
+        try {
+            recipientRewriteTable.addAliasMapping(source, aliasSourceAddress.asString());
+        } catch (MappingAlreadyExistsException e) {
+            // ignore
+        }
+    }
+
+    private void ensureUserDoesNotExist(MailAddress mailAddress) throws UsersRepositoryException {
+        if (usersRepository.contains(mailAddress.asString())) {
+            throw ErrorResponder.builder()
+                .statusCode(HttpStatus.BAD_REQUEST_400)
+                .type(ErrorResponder.ErrorType.INVALID_ARGUMENT)
+                .message("The alias source exists as an user already")
+                .haltError();
+        }
+    }
+
+    private MailAddress parseMailAddress(String address) {
```

Comment:
This piece of code is duplicated in `ForwardRoutes` & `GroupRoutes`.

Maybe we can introduce a `MailAddressParser` with a static method `parseMailAddress` that would be called in these 3 places? (a potential "addressType" parameter would help to keep the same message)